lib/policy_machine_storage_adapters/active_record/postgresql.rb

```diff
@@ -2,6 +2,95 @@
 
 module PolicyMachineStorageAdapter
   class ActiveRecord
+    class PolicyElementAssociation
+      def self.all_accessible_objects(associations, root_id: nil)
+        query = <<-SQL
+          id IN (
+            WITH RECURSIVE ancestors AS (
+            (
+              SELECT
+                objects.id AS parent_id
+                ,objects.id AS child_id
+              FROM policy_element_associations peas
+              JOIN policy_elements objects
+                ON objects.id = peas.object_attribute_id
+              WHERE peas.id IN (#{associations.select(:id).to_sql})
+            )
+            UNION ALL
+            (
+              SELECT
+                a.parent_id
+                ,a.parent_id
+              FROM assignments a
```

Comment:
your aliasing is kind of inconsistent

Reply:
fair enough, i'll update. lemme know if you think it's still unclear/inconsistent